js/customize-partial-refresh-settings.js

```diff
@@ -0,0 +1,128 @@
+/* global jQuery, _customizePartialRefreshSettings, JSON, _ */
+/* exported customizePartialRefreshSettings */
+var customizePartialRefreshSettings = ( function( $ ) {
+	'use strict';
+
+	var self, api = wp.customize;
+
+	self = {
+		setting: null
+	};
+
+	$.extend( self, _customizePartialRefreshSettings );
+
+	/**
+	 * Inject the functionality.
+	 */
+	self.init = function() {
+		api.bind( 'ready', function() {
+			self.onChangeSetting();
+		} );
+
+		api.bind( 'setting-partial', function( response ) {
+			var that = $( 'iframe' ).contents(),
+				value = $.trim( response );
+
+			$( self.setting.selector, that ).html( value );
```

Comment:
I like how you're fetching the Ajax for the partial in the Customizer pane, since this means you have access to the dirty settings. 

One thing to change here is that this should actually be doing `api.previewer.send()` instead of accessing the iframe's contents directly. There are situations where cross-domain security policy will make this cross-window communication forbidden, so it should be sending messages like how the Customizer communicates elsewhere.